api/interaction.go

```diff
@@ -25,6 +25,7 @@ const (
 	DeferredMessageUpdate
 	UpdateMessage
 	AutocompleteResult
+	Modal
```

Comment:
Can this be named `ModalResponse`? `Modal` is a bit too ambiguous.

(As a side note, I think it's worth renaming these constants to have the `ResponseType` suffix, so `ModalResponseType`. I'll consider it after this PR.)

Reply:
Yeah I was a bit stuck on the naming so I just decided to go with what Discord named it, and I also agree with renaming the constants to `xResponseType`, It would make it easier to know what to look for IMO.